kernel/hardware/interrupts.cpp

```diff
@@ -180,12 +186,14 @@ class ContextSwitcher {
 };  // class ContextSwitcher
 
 extern "C" void exception_handler(InterruptSource source, InterruptKind kind, Registers& registers) {
-  if (source == InterruptSource::CURRENT_SP_ELX && kind == InterruptKind::SYNCHRONOUS) {
```

Comment:
Kernel Synchronous Interrupt must be processed prior any allocation possible (In case of a Data Abort, we want to retrieve it !)
I don't known if all the call made by the `ContextSwitcher` do not allocate memory but in both case, it's a pain to check & maintains. Knowing that we will panic on a synchronous interrupt in the kernel, I prefer to process them first and panic quickly to not have the time to make other errors... (`do_kernelspace_interrupt` modification are also discussed here)